ngraph/core/src/op/reduce_logical_and.cpp

```diff
@@ -62,16 +55,25 @@ namespace
             return false;
         }
     }
-} // namespace
+} // namespace reduce_and
 
 bool op::v1::ReduceLogicalAnd::evaluate(const HostTensorVector& outputs,
                                         const HostTensorVector& inputs) const
 {
     NGRAPH_OP_SCOPE(v1_ReduceLogicalAnd_evaluate);
+    NGRAPH_CHECK(validate_host_tensor_vector(inputs, 2));
+    NGRAPH_CHECK(validate_host_tensor_vector(outputs, 1));
     const auto& data = inputs[0];
     const auto& axes = inputs[1];
     const auto& out = outputs[0];
-    return evaluate_reduce_logical_and(data, axes, out, get_keep_dims());
+    if (data->get_element_type() != element::boolean ||
```

Comment:
I think you can just use `is_integral()` here since it also returns true for booleans.

Reply:
But checking `data ` input for `is_integral()` would allow integer element types for the input, which for logical reduce operations is forbidden based on current [spec ](https://github.com/openvinotoolkit/openvino/pull/5612/files#diff-2e4d9a4ca174d6a600cd347c59c0e695c63e257ef1e08778ddf08ba0b0fb810aR45)

Reply:
Ok, in that case you should only check if `data->get_element_type() != element::boolean` in this `if`.

Reply:
My bad, the second check is for axes. You can disregard it, sorry for the noise ;)